lib/reek/smell_detectors/module_initialize.rb

```diff
@@ -21,14 +21,16 @@ def self.contexts # :nodoc:
       # @return [Array<SmellWarning>]
       #
       def sniff
-        context.defined_instance_methods.each do |node|
```

Comment:
Apparently, the cause of #1505 is that `ModuleContext#defined_instance_methods` includes methods from dynamically defined nested classes. Instead of dropping down to bare sexp manipulation, I think this method should be adjusted instead.

Reply:
Do you have any clue where can I start?